dogu/interfaces.go

```diff
@@ -26,3 +26,59 @@ type LocalRegistry interface {
 type configMapClient interface {
 	corev1client.ConfigMapInterface
 }
+
+type SimpleDoguName string
+
+// in common lib
+type DoguVersion struct {
+	Name    SimpleDoguName
+	Version core.Version
+}
+
+type DoguVersionRegistry interface {
+	GetCurrent(context.Context, SimpleDoguName) (DoguVersion, error)
+	GetCurrentOfAll(context.Context) ([]DoguVersion, error)
+	IsEnabled(context.Context, DoguVersion) (bool, error)
+	Enable(context.Context, DoguVersion) error
+	WatchAllCurrent(context.Context) (CurrentVersionsWatch, error)
+}
+
+type CurrentVersionsWatch struct {
+	ResultChan <-chan CurrentVersionsWatchResult
+	cancelFunc context.CancelFunc
+}
+
+type CurrentVersionsWatchResult struct {
+	Versions     map[SimpleDoguName]core.Version
+	PrevVersions map[SimpleDoguName]core.Version
+	Diff         []DoguVersion
+	Err          error
+}
+
+// LocalDoguDescriptorRepository is an append-only Repository, no updates will happen
+type LocalDoguDescriptorRepository interface {
+	// NotFoundError if dogu descriptor does not exist
+	Get(context.Context, DoguVersion) (*core.Dogu, error)
+	GetAll(context.Context, []DoguVersion) map[DoguVersion]*core.Dogu
+	// Add inserts a new dogu descriptor.
+	// ConflictError if a dogu descriptor with this dogu name and version already exists
+	// ConnectionError if there are any connection issues
+	// a generic error at any other error
+	Add(context.Context, SimpleDoguName, *core.Dogu) error
+	// Delete is currently not used and probably unnecessary
+	// Delete(context.Context, SimpleDoguName, core.Version) error
+	DeleteAll(context.Context, SimpleDoguName) error
+	// Do we need this? We can just watch for a new dogu version and then pull the dogu descriptor
+	// WatchAll(context.Context, SimpleDoguName) (DoguWatch, error)
+}
+
+type DoguWatch struct {
+	ResultChan chan<- DoguWatchResult
+	cancelFunc context.CancelFunc
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>field `cancelFunc` is unused
<!-- __reviewdog__:ChAxOWVkMmNiYzdiOGVlZDUzEgpjaGVja3N0eWxl -->